gateway-js/src/index.ts

```diff
@@ -346,6 +359,36 @@ export class ApolloGateway implements GraphQLService {
     }
   }
 
+  private issueDeprecationWarningsIfApplicable() {
```

Comment:
Would it be feasible to change all of the deprecated options to be implemented in terms of the new functionality?  ie serviceList (and localServiceList? not sure what that is) could be turned into IntrospectAndCompose, and the other two functions could be mapped more directly.  If you did that then it would make it more clear that the new functionality is a superset of the old functionality...

Reply:
Done, over a range of commits (let me know if it's helpful to specify, I imagine you'll find them though)